src/note.rs

```diff
@@ -0,0 +1,48 @@
+macro_rules! parse_env {
+	($name:expr $(,)?) => {
+		::const_parse::parse_u128(::core::env!($name)) as u32
+	};
+}
+
+#[macro_export]
+macro_rules! define_abi_tag {
+	() => {
+		#[used]
+		#[unsafe(link_section = ".note.ABI-tag")]
+		static ABI_TAG: $crate::note::_AbiTag = $crate::note::_AbiTag::new();
+	};
+}
+
+#[repr(C)]
+#[doc(hidden)]
+pub struct _AbiTag {
+	header: Nhdr32,
+	name: [u8; 8],
+	data: [u32; 4],
+}
+
+impl _AbiTag {
+	pub const fn new() -> Self {
+		Self {
+			header: Nhdr32 {
+				n_namesz: 7,
+				n_descsz: 1,
+				n_type: 1,
```

Comment:
I'm not too deep into elf headers and even if this field does matter on a note header, but for the hermit-entry version field is set to `0x5a00`.